### PR TITLE
feat(compose): run devcontainer feature entrypoints on the compose path

### PR DESCRIPTION
**Pull request by eve0415**
## What

Devcontainer **feature entrypoints** now run on container start for docker-compose projects, matching the official devcontainer CLI. Until now only the single-container path chained them; on compose they were silently dropped, so a feature that relied on its `entrypoint` (e.g. starting a daemon, fixing perms) just didn't run.

The compose override now emits a wrapped entrypoint that runs each feature entrypoint in install order, then `exec`s the service's original entrypoint+command:

```yaml
    entrypoint: ["/bin/sh", "-c", "echo Container started\ntrap \"exit 0\" 15\n<feat-a ep>\n<feat-b ep>\nexec \"$$@\"\nwhile sleep 1 & wait $$!; do :; done", "-", "/docker-entrypoint.sh"]
```

This mirrors `devcontainers/cli` `dockerCompose.ts` — including the bits that are easy to get wrong:

- **`$$@` / `$$!` are literal double-dollars.** The override is a `-f` compose file, so `docker compose` interpolates `$`; doubling makes the shell receive `$@` / `$!`. Image-derived entrypoint/cmd values are `$`→`$$` escaped too; service-file values come from the already-resolved `docker compose config` (which re-escapes `$`→`$$`) so they're emitted verbatim.
- **No agent restart loop here.** The single-container entrypoint bakes in a `cella-agent daemon` restart loop; on compose the agent is launched separately via `nohup` after `up`, so the compose entrypoint must NOT include it. It only runs feature entrypoints + `exec` + keepalive.
- **`userEntrypoint`/`userCommand` resolution** matches the official `||` fallback exactly: `overrideCommand` discards both; otherwise the service's `entrypoint`/`command` win, falling back to the image's `Config.Entrypoint`/`Cmd`; when the service overrides the entrypoint the image CMD is ignored (compose spec); and `command:` is emitted only when it differs from the compose-declared command.

## How it's built

- `feat(backend)`: `ImageDetails` surfaces the image's `entrypoint`/`cmd` (needed for the fallback above). Pure `image_details_from_inspect` helper, unit-tested against real bollard types. Apple `container` backend reports `compose: false`, so its values default to empty (compose is docker-only).
- `feat(compose)`: extract the service `entrypoint`/`command` from the resolved compose config (string → shell-split, array → verbatim, absent → none). Pulls `shell-words` up to a pinned workspace dep.
- `feat(compose)`: `OverrideConfig` gains `feature_entrypoints` / `user_entrypoint` / `user_command`; `generate_override_yaml` emits the wrapped block **only** when there are feature entrypoints OR `overrideCommand` — so a no-feature, non-override service's override is byte-for-byte unchanged. Resolution lives in a pure `resolve_user_entrypoint_command` (faithful port, incl. the empty-array-is-truthy edge cases) plus a `resolve_compose_user_entrypoint_command` that reads the *un-overridden* config so it sees the user's original entrypoint, not cella's wrapper.
- `test(compose)`: a `runtime_test(docker, compose)` builds a feature whose entrypoint drops a sentinel, brings the service up, and asserts both that the sentinel exists (the entrypoint ran) and the base command is still running (the `exec` passthrough works). Skips locally, runs in CI.
- `refactor(compose)`: short-circuit the config+inspect when `overrideCommand` makes the result fixed (matches the official's lazy image inspect).

## Notes

- 11 emitter/resolver unit tests + the integration test cover escaping, the image fallback, the userEntrypoint passthrough, the conditional `command:`, the no-feature no-op, and `overrideCommand` discard.
- Full spec lives in `.claude/plans/compose-feature-entrypoints.md` (local).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/cella-compose/src/orchestrate.rs [hunk 1]
```diff
@@ -1075,6 +1111,78 @@ async fn resolve_compose_image_info(
...
 
+/// Resolve the wrapped entrypoint's `userEntrypoint`/`userCommand` for the
+/// primary service, mirroring the official compose logic.
+///
+/// Reads the service's resolved `entrypoint`/`command` from `docker compose
+/// config` and the base image's `ENTRYPOINT`/`CMD` from an inspect, then applies
+/// [`crate::override_file::resolve_user_entrypoint_command`]. Falls back to the
+/// default (empty entrypoint, no command) when the compose config can't be
+/// resolved — the wrapped entrypoint then just runs feature entrypoints and
+/// `exec "$@"` over whatever the service already had.
```
[the remark this answers is not in the document]
Comment (eve0415) [reply crates/cella-compose/src/orchestrate.rs]: Good catch — fixed in 27a226c. You're right: the wrapped entrypoint replaces the service's own, so defaulting to an empty `userEntrypoint` on a resolution failure would silently drop the original ENTRYPOINT and `exec "$@"` only the command — starting the container with the wrong process.

Now the `docker compose config` and image-inspect failures propagate instead of defaulting (the official CLI hard-errors here too — its config/`imageDetails` are required). I also made the image inspect lazy — it only runs when the service declares no `entrypoint`, the single case where the image fallback is actually used, matching the official `composeEntrypoint || imageEntrypoint` short-circuit. The misleading doc comment is corrected too.

File: crates/cella-compose/src/build_features.rs [hunk 2]
```diff
@@ -132,6 +132,12 @@ pub async fn compose_build(
             request_gpu: false,
             // `cella build` only builds the image; runtime security props apply at `up`.
             security: cella_config::config_map::MergedSecurityConfig::default(),
+            // Feature entrypoints + entrypoint/command resolution belong to the
+            // runtime override (`write_final_override`); the build override never
+            // runs the container, and `docker compose build` ignores these keys.
+            feature_entrypoints: Vec::new(),
+            user_entrypoint: Vec::new(),
+            user_command: None,
```
[the remark this answers is not in the document]
Comment (eve0415) [reply crates/cella-compose/src/build_features.rs:146]: Good catch — fixed in f603ee8, taking your second suggestion (don't emit the runtime entrypoint from the build override).

The build override is purely a build artifact: `cella build` only builds the image, and `cella up` writes its own runtime override via `write_final_override` — nothing in cella runs the persisted build override. So I forced `override_command: false` there, which suppresses the entrypoint section entirely (the gate is `feature_entrypoints.is_empty() && !override_command`). For the same reason — it never runs the container — I also set `build_only: true` so it carries no agent volume, matching the labels-only build override. (That corrects an earlier comment of mine that wrongly claimed this override was "reused at up".)

Added a regression test locking the build-override shape: `build:` section present, no `entrypoint:`/`command:`/`volumes:`.
